### DISCUSSION TITLE
Questions about the TE/TM mode splitter in [High-performance hybrid time/frequency-domain topology optimization for large-scale photonics inverse design](https://opg.optica.org/oe/fulltext.cfm?uri=oe-30-3-4467&id=468836)

### DISCUSSION THREAD

**Discussion by niccolot** [NanoComp/meep#2810] [Q&A] [answered]
I wanted to ask a couple of questions about the mode splitter presented in the article as a first example of optimized structure. 

In the article is being said that 2 objective functions are used, one for the first TE mode and one for the first TM mode. In the definitions the coefficient belong to the first mode $\alpha_{1i}$ for the TE and to the second mode for the TM $\alpha_{2i}$. I wanted to ask if this corresponds in setting `mode=1` in the `mpa.EigenmodeCoefficient` (corresponding to `eig_band=1` in `mp.EigenModeSource`) and how  one can infer a priori that the first mode will be quasi TE and the second quasi TM, by using a mode solver first on the rectangular inlet waveguide profile?

![Screenshot 2024-03-27 122104](https://github.com/NanoComp/meep/assets/111637024/9058c2d5-822f-4d76-8581-49cd1421717f)
![Screenshot 2024-03-27 122050](https://github.com/NanoComp/meep/assets/111637024/9ce07eef-7d8f-457f-ae71-34792e422ff0)

The second question is regarding the plots: for the quasi TE is being plotted Hz, for the quasi TM ez with a propagation direction along x. I wanted to ask why is that since naively i would have done the opposite.

![Screenshot 2024-03-27 122124](https://github.com/NanoComp/meep/assets/111637024/00121305-28d2-4e9e-b3f4-33fa0710bf2e)

Thank you in advance.

**Comment by stevengj** [comment]
> how one can infer a priori that the first mode will be quasi TE and the second quasi TM

quasi-TE and quasi-TM actually correspond to odd/even modes with respect to a mirror symmetry plane bisecting the waveguide, so you can specify them with the parity argument.   Then for each "polarization" it is just the fundamental mode.

**Comment by smartalecH** [comment, marked as answer]
> The second question is regarding the plots: for the quasi TE is being plotted Hz, for the quasi TM ez with a propagation direction along x. I wanted to ask why is that since naively i would have done the opposite.

If you plot the mode profiles from MPB, you'll notice in which component most of the energy is stored. There are a lot of confusing (and contradicting) definitions of TE vs TM. Which is why we plotted the fields in the paper (to make it clear what we meant). But in our case, TE means most of the E-field energy is in plane, meaning most of the H-field energy is out of plane (so we plot Hz, because it's easier than plotting both Ex and Ey). 

In fact, you can use this trick to figure which fundamental polarization corresponds to which mode index (if you don't use the parity trick Steven mentioned). 

In short, just look at the mode fields themselves.
